s3connector/s3_mock.go

```diff
@@ -0,0 +1,78 @@
+package s3connector
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+)
+
+type S3Mock struct {
```

Comment:
Why do we need mock once we have localstack?

Reply:
This mock was used on the ingesters so I've copied the logic for further use if needed.